src/components/TopologySidebar/TopologyActionBar.tsx

```diff
@@ -0,0 +1,286 @@
+          icon={icon}
+          text={text}
+        />
+      );
+    }
+  },
+  {
+    label: "Zoom In",
+    Icon: BiZoomIn,
+    isShown: (topology, displayPoint) => displayPoint === "TopologyCard",
+    ContainerComponent: function Container({
+      child: ChildComponent,
+      topology,
+      icon,
+      text
+    }) {
+      const navigate = useNavigate();
+
+      return (
+        <ChildComponent
+          onClick={() => navigate(`/topology/${topology.id}`)}
+          icon={icon}
+          text={text}
+        />
+      );
+    }
+  },
+  {
+    label: "Link to Incident",
+    Icon: MdAlarmAdd,
+    isShown: () => true,
+    ContainerComponent: function Container({
+      child: ChildComponent,
+      topology,
+      icon,
+      text,
+      openModalAction: onModalOpen = () => {}
+    }) {
+      return (
+        <ChildComponent onClick={() => onModalOpen()} icon={icon} text={text} />
+      );
+    }
+  },
+  {
+    label: "Hide",
+    Icon: BiHide,
+    isShown: (topology) => topology.hidden !== false,
+    ContainerComponent: function Container({
+      child: ChildComponent,
+      topology,
+      onRefresh,
+      icon,
+      text
+    }) {
+      const { mutate: updateVisibility } =
+        useUpdateComponentMutation(onRefresh);
+
+      return (
+        <ChildComponent
+          onClick={() =>
+            updateVisibility({
+              id: topology.id,
+              hidden: true
+            })
+          }
+          icon={icon}
+          text={text}
+        />
+      );
+    }
+  },
+  {
+    label: "Show",
+    Icon: BiShow,
+    isShown: (topology) => topology.hidden === false,
+    ContainerComponent: function Container({
+      child: ChildComponent,
+      topology,
+      onRefresh,
+      icon,
+      text
+    }) {
+      const { mutate: updateVisibility } =
+        useUpdateComponentMutation(onRefresh);
+
+      return (
+        <ChildComponent
+          onClick={() =>
+            updateVisibility({
+              id: topology.id,
+              hidden: false
+            })
+          }
+          icon={icon}
+          text={text}
+        />
+      );
+    }
+  },
+  {
+    label: "Snapshot",
+    Icon: ImTree,
+    isShown: () => true,
+    ContainerComponent: function Container({
+      child: ChildComponent,
+      topology,
+      icon,
+      text,
+      openModalAction: onModalOpen = () => {}
+    }) {
+      return (
+        <ChildComponent onClick={() => onModalOpen()} icon={icon} text={text} />
+      );
+    }
+  }
+];
+
+type TopologyActionLinkProps = {
+  onClick?: () => void;
+  icon: React.ReactNode;
+  text: React.ReactNode;
+};
+
+function TopologyActionLink({
+  onClick = () => {},
+  icon,
+  text
+}: TopologyActionLinkProps) {
+  return (
+    <div className="flex w-full flex-col items-center p-1 h-full">
+      <Button
+        icon={icon}
+        text={text}
+        className="flex w-full h-full flex-col space-y-2 justify-center items-center btn-primary"
```

Comment:
We should not be changing the class on the buttons - the only thing that should every change is the color. The style should be normal button icon on the left, text on the right - rectangle, not square